pkg/controller/user-cluster-controller-manager/resources/resources/envoy-agent/daemonset.go

```diff
@@ -0,0 +1,181 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package envoyagent
+
+import (
+	"fmt"
+	"net"
+
+	"k8c.io/kubermatic/v2/pkg/resources"
+	"k8c.io/kubermatic/v2/pkg/resources/reconciling"
+	"k8c.io/kubermatic/v2/pkg/version/kubermatic"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	utilpointer "k8s.io/utils/pointer"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	defaultResourceRequirements = map[string]*corev1.ResourceRequirements{
+		resources.EnvoyAgentDaemonSetName: {
+			Requests: corev1.ResourceList{
+				corev1.ResourceMemory: resource.MustParse("32Mi"),
+				corev1.ResourceCPU:    resource.MustParse("50m"),
+			},
+			Limits: corev1.ResourceList{
+				corev1.ResourceMemory: resource.MustParse("64Mi"),
+				corev1.ResourceCPU:    resource.MustParse("100m"),
+			},
+		},
+	}
+)
+
+const (
+	envoyImageName = "envoyproxy/envoy"
+	envoyImageTag  = "v1.17.0"
+)
+
+// DaemonSetCreator returns the function to create and update the Envoy DaemonSet
+func DaemonSetCreator(agentIP net.IP, versions kubermatic.Versions) reconciling.NamedDaemonSetCreatorGetter {
+	return func() (string, reconciling.DaemonSetCreator) {
+		return resources.EnvoyAgentDaemonSetName, func(ds *appsv1.DaemonSet) (*appsv1.DaemonSet, error) {
+			ds.Name = resources.EnvoyAgentDaemonSetName
+			ds.Namespace = metav1.NamespaceSystem
+			ds.Labels = resources.BaseAppLabels(resources.EnvoyAgentDaemonSetName, nil)
+
+			ds.Spec.Selector = &metav1.LabelSelector{
+				MatchLabels: resources.BaseAppLabels(resources.EnvoyAgentDaemonSetName,
+					map[string]string{"app.kubernetes.io/name": "envoy-agent"}),
+			}
+
+			// has to be the same as the selector
+			ds.Spec.Template.ObjectMeta = metav1.ObjectMeta{
+				Labels: resources.BaseAppLabels(resources.EnvoyAgentDaemonSetName,
+					map[string]string{"app.kubernetes.io/name": "envoy-agent"}),
+			}
+
+			//TODO(youssefazrak) needed?
+			ds.Spec.Template.Spec.PriorityClassName = "system-cluster-critical"
+			ds.Spec.Template.Spec.DNSPolicy = corev1.DNSClusterFirst
+
+			ds.Spec.Template.Spec.HostNetwork = true
+
+			volumes := getVolumes()
+			ds.Spec.Template.Spec.Volumes = volumes
+
+			ds.Spec.Template.Spec.InitContainers = getInitContainers(agentIP, versions)
+			ds.Spec.Template.Spec.Containers = getContainers()
+			err := resources.SetResourceRequirements(ds.Spec.Template.Spec.Containers, defaultResourceRequirements, nil, ds.Annotations)
+			if err != nil {
+				return nil, fmt.Errorf("failed to set resource requirements: %v", err)
+			}
+
+			ds.Spec.Template.Spec.ServiceAccountName = resources.CoreDNSServiceAccountName
```

Comment:
Do we need a specific SA? If not we should remove this line (and copy/paste forgotten value assigned to coredns' SA)

Reply:
nope, we should remove this.